Tasks/Common/func-tools-common/functoolsutility.ts

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as util from 'util';
+import * as tl from "azure-pipelines-task-lib/task";
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as path from "path";
+
+const funcToolName = 'func';
+
+function getExecutableExtension(): string {
+    if (os.type().match(/^Win/)) {
+        return '.exe';
+    }
+
+    return '';
+}
+
+function getDownloadUrl(version: string) {
+    switch (os.type()) {
+        case 'Linux':
+            return util.format('https://github.com/Azure/azure-functions-core-tools/releases/download/%s/Azure.Functions.Cli.linux-x64.%s.zip', version, version);
+
+        case 'Darwin':
+            return util.format('https://github.com/Azure/azure-functions-core-tools/releases/download/%s/Azure.Functions.Cli.osx-x64.%s.zip', version, version);
+
+        case 'Windows_NT':
+        default:
+            return util.format('https://github.com/Azure/azure-functions-core-tools/releases/download/%s/Azure.Functions.Cli.win-x86.%s.zip', version, version);
+
+    }
+}
+
+export async function getLatestFuncToolsVersion(): Promise<string> {
+    const funcToolsLatestReleaseUrl = 'https://api.github.com/repos/Azure/azure-functions-core-tools/releases/latest';
+    const stableFuncToolsVersion = '2.7.1585';
+    let latestVersion = stableFuncToolsVersion;
+
+    try {
+        const downloadPath = await toolLib.downloadTool(funcToolsLatestReleaseUrl);
```

Comment:
You could have just done a REST call using the `typed-rest-client` instead, right? Why choose this path?

Reply:
It handles redirection and retries natively. Also, we are using it in multiple places - kubectl, helm etc.